src/shared/components/Input/Input.jsx

```diff
@@ -1,31 +1,55 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-function Input({ name, className, placeholder, onChange, type }) {
+function Input({
+  value,
+  className,
+  inputRef,
+  placeholder,
+  onChange,
+  type,
+  onBlur,
+}) {
   return (
     <input
       className={className}
       placeholder={placeholder}
-      name={name}
+      value={value}
       type={type}
       onChange={onChange}
+      ref={inputRef}
+      onBlur={onBlur}
     />
   );
 }
 Input.propTypes = {
   className: PropTypes.string,
   placeholder: PropTypes.string,
   onChange: PropTypes.func,
-  name: PropTypes.string,
+  onBlur: PropTypes.func,
+  value: PropTypes.string,
   type: PropTypes.string,
+  inputRef: PropTypes.objectOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+    }),
+  ),
 };
 
 Input.defaultProps = {
   className: PropTypes.string,
   placeholder: PropTypes.string,
   onChange: PropTypes.func,
-  name: PropTypes.string,
+  onBlur: PropTypes.func,
+  value: PropTypes.string,
   type: PropTypes.string,
+  inputRef: PropTypes.objectOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+    }),
```

Comment:
I don't think this correct the structure for ref.

Reply:
Fixed